Kickstarter-iOS/Views/Controllers/DiscoveryFiltersViewController.swift

```diff
@@ -200,10 +201,11 @@ internal final class DiscoveryFiltersViewController: UIViewController, UITableVi
   }
 
   private func deleteCategoriesLoaderRow() {
-    self.filtersTableView.beginUpdates()
+    guard let deleteCategoriesLoaderRow = self.dataSource.deleteCategoriesLoaderRow() else { return }
 
-    self.filtersTableView.deleteRows(at: self.dataSource.deleteCategoriesLoaderRow(), with: .fade)
+    self.filtersTableView.beginUpdates()
+    defer { self.filtersTableView.endUpdates() }
 
-    self.filtersTableView.endUpdates()
+    self.filtersTableView.deleteRows(at: deleteCategoriesLoaderRow, with: .fade)
```

Comment:
does `deleteRows` crash if you use an empty array?

I bet it does. cause table views are the worst :/